arches_component_lab/src/arches_component_lab/widgets/FileListWidget/components/FileListWidgetEditor.vue

```diff
@@ -0,0 +1,199 @@
+<script setup lang="ts">
+import { onMounted, useTemplateRef, ref, watch, computed } from "vue";
+import { useGettext } from "vue3-gettext";
+
+import Message from "primevue/message";
+import FileUpload, {
+    type FileUploadRemoveEvent,
+    type FileUploadSelectEvent,
+} from "primevue/fileupload";
+import Image from "primevue/image";
+import Button from "primevue/button";
+import { FormField } from "@primevue/forms";
+
+import type { FormFieldResolverOptions } from "@primevue/forms";
+import type { FileReference } from "@/arches_component_lab/widgets/types.ts";
+const { $gettext } = useGettext();
+const allowedFileTypes = ref();
+const props = defineProps<{
+    initialValue: FileReference[] | null | undefined;
+    graphSlug: string;
+    nodeAlias: string;
+    widgetData: {
+        config: {
+            acceptedFiles: string;
+            maxFilesize: number;
+            rerender: boolean;
+            label: string;
+        };
+    };
+    nodeData: {
+        config: {
+            imagesOnly: boolean;
+            maxFiles: number;
+            maxFileSize: number;
+        };
+    };
+}>();
+
+const formFieldRef = useTemplateRef("formFieldRef");
+const currentValues = ref<FileReference[]>();
+
+watch(
+    () => props.initialValue,
+    (newValue) => {
+        if (newValue) {
+            currentValues.value = newValue;
+        } else {
+            currentValues.value = [];
+        }
+    },
+);
+
+onMounted(() => {
+    allowedFileTypes.value = props.nodeData.config.imagesOnly ? "image/*" : "*";
```

Comment:
why not widgetData.config.acceptedFiles? Also should it be changed to restrict any filetype?

Reply:
yeah there's a couple of things at play here. I _do_ think we should centralize to a single source of truth for widgets, and it should be the widget config. That said, I can understand the need for a separate node config that defines how the node will behave/what it will accept regardless of widget. But still, IMO this should be copied into card_x_node_x_widget configs.

The bit I'm taking issue with here is using `props.nodeData.config.imagesOnly` for control flow. What happens if I only want to accept .txt files? It seems like this widget is being configured for only images or everything.

Reply:
I'm fine with this either way and if widgetData's the answer I'm happy to use that instead.  Good point re: text files.  This is fixed.  Amusingly enough, it wasn't even being used.  See file uploader change.